cowprotocol/cowswap
# Fix issue signing using WC for some wallets

# Summary
This PR fixes the issue with 1inch wallet. Resullt of the review #496 

## Short description
This PR fixes the issue where, when trading, the user was prompted to sign only a hash instead of the JSON with all the order details.

This was happening to 1inch wallet, and possibly other wallets too.

Before, in 1inch wallet
![IMG_5904](https://user-images.githubusercontent.com/2352112/166925886-2f93ab04-70d6-4c93-ace9-880f39509816.PNG)

After, in 1inch wallet
![IMG_5905](https://user-images.githubusercontent.com/2352112/166925868-2682e5b0-56e6-4200-abfd-76baff038a9d.PNG)


## Short technical description
Uses a new version of the smart contracts which allows us to use `eth_signTypedData` instead of `eth_signTypedData_v4` for signing.


## Longer explanation
The issue was making the signing method to default to ETHSIGN instead of using EIP-712

This is because the wallet was not understanding the signing method we provide with the request. And the app tries to downgrade to v3 and ETHSIGN if this is the case.

The actual problem is that by upgrading web3-react, a new wallet-connect version came with it. This version was using a different wallet connect provider (@walletconnect/ethereum-provider instead of @walletconnect/web3-provider).

This shouldn't be a big deal, because they generate the same signing request. However, at some point for the old version, wallet connect was replacing the signing method name from `eth_signTypedData_v4` to just `eth_signTypedData`.

Apparently 1inch wallet understand the latter but not the former.


## Note for the reviewers
Please excuse me for adding all the new org changes for the NPM package. 

Allow me this in this case. I can point out to the only 2 changes, please see the comments in the code.

# To Test

1. Sign an order with 1inch wallet in IOS.
2. Verify you see the JSON details with all the order fields
